hedera-node/docs/design/services/smart-contract-service/frictionless-airdrops-system-contracts.md

```diff
@@ -0,0 +1,164 @@
+- `RejectTokensTranslator` - This class will be responsible for handling the `rejectTokens`, `rejectTokenFT` and `rejectTokenNFT` selectors and dispatching them to the corresponding HAPI calls.
+- `RejectTokensDecoder` - This class provides methods and constants for decoding the given `HtsCallAttempt` into `TokenReference` list for `TokenReject` call.
+- `GetPendingAirdropsTranslator` - This class will be responsible for handling the `getPendingAirdrops` selector for querying the pending airdrop store.
+- `GetPendingAirdropsCall` - This class will be responsible for encoding and dispatching the response data.
+
+## Security Implications
+
+The newly added flows will adopt the HAPI authorization logic and the security V2 model.
+We will apply the `TokenReject`, `TokenAirdrop`, `TokenClaimAirdrop`, `TokenCancelAirdrop` throttle mechanisms.
+
+## Acceptance Tests
+
+### BDD Tests
+
+#### Positive Tests
+
+- Verify that the `airdropTokens` function airdrops multiple tokens bot ft and nft to multiple accounts.
+- Verify that the `airdropTokens` function airdrops a fungible token to an account.
+- Verify that the `airdropTokens` function airdrops a nft token to an account.
+- Verify that the `cancelAirdrops` function cancels multiple pending airdrops.
+- Verify that the `cancelAirdrops` function cancels single pending airdrop.
+- Verify that the `claimAirdrops` function claims multiple pending airdrops.
+- Verify that the `claimAirdrops` function claims a single pending airdrop.
+- Verify that the `rejectTokens` function rejects tokens for multiple accounts.
+- Verify that the `getPendingAirdrops` function returns the pending airdrops for a given account.
+- Verify that the `cancelAirdropFT` function cancels a pending airdrop of the redirected token.
+- Verify that the `cancelAirdropNFT` function cancels a pending airdrop of the redirected nft serial.
+- Verify that the `claimAirdropFT` function claims a pending airdrop of the redirected token.
+- Verify that the `claimAirdropNFT` function claims a pending airdrop of the redirected nft serial number.
+- Verify that the `rejectTokenFT` function rjects tokens for a given account.
+- Verify that the `rejectTokensNFT` function rejects tokens for a given account and serial number.
+- Verify that the `setAutomaticAssociations` function sets the automatic associations for a given account.
+- Verify that the `setAutomaticAssociations` function sets the automatic associations to unlimited (-1) for a given account.
+- Verify that the `setAutomaticAssociations` function sets the automatic associations to zero for a given account.
+
+#### Negative Tests
+
+- Verify that the `airdropTokens` function fails when the sender does not have enough balance.
+- Verify that the `airdropTokens` function fails when the receiver does not have a valid account.
+- Verify that the `airdropTokens` function fails when the token does not exist.
```

Comment:
Also validate that out of bounds values for the transfer list fails

Reply:
Thanks, I'll add more negative cases for out of bounds values